routing/cross_mwm_index_graph.cpp

```diff
@@ -12,8 +14,33 @@ using namespace std;
 
 namespace
 {
-bool GetTransitionSegment(OsrmFtSegMapping const & segMapping, TWrittenNodeId nodeId,
-                          NumMwmId numMwmId, Segment & segment)
+/// \returns FtSeg by |segment|.
+OsrmMappingTypes::FtSeg GetFtSeg(Segment const & segment)
+{
+  uint32_t const startPnt = segment.IsForward() ? segment.GetPointId(false /* front */)
+                                                : segment.GetPointId(true /* front */);
+  uint32_t const endPnt = segment.IsForward() ? segment.GetPointId(true /* front */)
+                                              : segment.GetPointId(false /* front */);
+  return OsrmMappingTypes::FtSeg(segment.GetFeatureId(), startPnt, endPnt);
+}
+
+/// \returns a pair of direct and reverse(backward) node id by |segment|.
+/// The first member of the pair is direct node id and the second is reverse node id.
```

Comment:
Maybe it's worth to create a simple struct with two fields: m_directNodeId, m_reverseNodeId?